lib/ecto/adapters/clickhouse.ex

```diff
@@ -214,7 +220,7 @@ defmodule Ecto.Adapters.ClickHouse do
           [{:command, :delete} | opts]
       end
 
-    result = Ecto.Adapters.SQL.query!(adapter_meta, sql, params, opts)
+    result = Ecto.Adapters.SQL.query!(adapter_meta, sql, params, put_source(opts, query_meta))
```

Comment:
`Ecto.Adapters.SQL`'s `put_source` wasn't called since `execute` is overwritten in this adapter. `execute` is overwritten to enable `{param:Type}` in SQL since this is the first place where we get access to parameters.